src/ccache.cpp

```diff
@@ -865,124 +865,104 @@ process_preprocessed_file(Context& ctx,
 static void
 use_relative_paths_in_depfile(const Context& ctx)
 {
-  const char* depfile = ctx.args_info.output_dep.c_str();
-
   if (ctx.config.base_dir().empty()) {
     cc_log("Base dir not set, skip using relative paths");
     return; // nothing to do
   }
   if (!ctx.has_absolute_include_headers) {
     cc_log(
-      "No absolute path for included files found, skip using relative"
-      " paths");
+      "No absolute path for included files found, skip using relative paths");
     return; // nothing to do
   }
 
-  FILE* f;
-  f = fopen(depfile, "r");
-  if (!f) {
-    cc_log("Cannot open dependency file: %s (%s)", depfile, strerror(errno));
+  std::string dependencies;
+  try {
+    dependencies = Util::read_file(ctx.args_info.output_dep);
+  } catch (const Error& e) {
+    cc_log("Cannot open dependency file %s: %s",
+           ctx.args_info.output_dep.c_str(),
+           e.what());
```

Comment:
Is it possible to add a test for this error message? You were able to change it a few commits ago and now again without adapting tests. So it can get broken just as easily.

Reply:
That level of testing is unfortunately still far away, so don't worry about.

But if there were a test, how would it have to be adapted? Isn't the message still exactly the same?